libmproxy/console/common.py

```diff
@@ -277,26 +277,73 @@ def copy_flow_format_data(part, scope, flow):
     return data, False
 
 
-def copy_flow(part, scope, flow, master, state):
-    """
-    part: _c_ontent, _h_eaders+content, _u_rl
-    scope: _a_ll, re_q_uest, re_s_ponse
-    """
-    data, err = copy_flow_format_data(part, scope, flow)
+def export_prompt(k, flow):
+    if k == "c":
+        copy_as_curl_command(flow)
+    elif k == "p":
+        copy_as_python_code(flow)
+    elif k == "r":
+        copy_as_raw_request(flow)
 
-    if err:
-        signals.status_message.send(message=err)
+
+def copy_as_curl_command(flow):
+    if flow.request.content is None or flow.request.content == CONTENT_MISSING:
```

Comment:
Um, to be quite honest, I don't totally get this line. It was already a part of pull #799 and since I couldn't think of a reason to remove it either. I just kept it. [Line](https://github.com/mitmproxy/mitmproxy/pull/799/files#diff-9641efb691a86154d6143b6ec654f8f7R281)